onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -902,6 +902,191 @@ ONNX_MS_OPERATOR_SET_SCHEMA(IsAllFinite, 1,
+                                " spatial axis, it can take any value greater than or equal to 0. "
+                                "The value represent the number of pixels added to the beginning "
+                                "and end part of the corresponding axis. `pads` format should be as follow "
+                                "[x1_begin, x2_begin...x1_end, x2_end,...], where xi_begin is the number of pixels "
+                                "added at the beginning of axis `i` and xi_end is the number of pixels added at the end of axis `i`. "
+                                "If not present, the padding defaults to 0 along start and end of each spatial axis.",
+                                AttributeProto::INTS,
+                                OPTIONAL_VALUE)
+                            .Attr(
+                                "strides",
+                                "1-dimensional tensor with stride value along each spatial axis. "
+                                "If not present, the stride defaults to 1 along each spatial axis.",
+                                AttributeProto::INTS,
+                                OPTIONAL_VALUE)
+                            .Input(
+                                0,
+                                "input",
+                                "Input data tensor to be rearranged from column blocks back into an image."
+                                " This is a 3-dimensional tensor containing [N, C * n-ary-product(block_shape), L],"
+                                " where N is batch dimension, C is image channel dimension and L is number of blocks.",
+                                "T",
+                                OpSchema::Single,
+                                true,
+                                1,
+                                OpSchema::Differentiable)
+                            .Input(
+                                1,
+                                "image_shape",
+                                "The shape of the spatial dimensions of the image after rearranging the column blocks."
+                                "This is a 1-dimensional tensor with size of at least 2, containing the value [H_img, W_img] "
+                                " for a 2-D image or [dim_i1, dim_i2, ..., dim_iN] for a N-D image.",
+                                "tensor(int64)",
+                                OpSchema::Single,
+                                true,
+                                1,
+                                OpSchema::NonDifferentiable)
+                            .Input(
+                                2,
+                                "block_shape",
+                                "The shape of the block to apply on the input."
+                                "This is a 1-dimensional tensor of size of at least 2, containing the value [H_block, W_block] "
```

Comment:
Dilation applies to this shape, right? Is the input supposed to be with or without dilation?

Reply:
No, it doesnt